src/functions-urls.php

```diff
@@ -2,21 +2,52 @@
 
 namespace Globalis\WP\Cubi;
 
+/**
+ * Get current url with query string or not
+ *
+ * @param  bool   $remove_query_args If true, will remove query string from url.
+ *
+ * @return string $url               The current url.
+ */
 function get_current_url($remove_query_args = false)
 {
-    $url = WP_SCHEME . '://' . WP_DOMAIN . $_SERVER['REQUEST_URI'];
-    if ($remove_query_args) {
-        $parts = parse_url($url);
-        $url   = WP_SCHEME . '://' . WP_DOMAIN . $parts['path'];
+    if (defined(WP_SCHEME) && defined(WP_DOMAIN)) {
+        $url = WP_SCHEME . '://' . WP_DOMAIN . $_SERVER['REQUEST_URI'];
+        if ($remove_query_args) {
+            $parts = parse_url($url);
+            $url   = WP_SCHEME . '://' . WP_DOMAIN . $parts['path'];
+        }
+    } else {
+        $protocol = isset($_SERVER['HTTPS']) ? 'https' : 'http';
+        $url = $protocol . '://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'];
```

Comment:
use something like `$host = ! empty($_SERVER['HTTP_X_FORWARDED_HOST']) ? $_SERVER['HTTP_X_FORWARDED_HOST'] : $_SERVER['HTTP_HOST'];` to catch reverse-proxy configurations limit cases